src/webgpu/api/operation/command_buffer/copies.spec.ts

```diff
@@ -1,33 +1,131 @@
 export const description = `
 copy{Buffer,Texture}To{Buffer,Texture} tests.
+
+Test Plan:
+* Validate the correctness of the copy by filling the srcBuffer with testable data, doing
+  CopyBufferToBuffer() copy, and verifying the content of the whole dstBuffer with MapRead:
+  Copy {4 bytes, part of, the whole} srcBuffer to the dstBuffer {with, without} a non-zero valid
+  srcOffset that
+  - covers the whole dstBuffer
+  - covers the beginning of the dstBuffer
+  - covers the end of the dstBuffer
+  - covers neither the beginning nor the end of the dstBuffer
+* Validate the state transitions after the copy:
+  first copy from srcBuffer to dstBuffer, then copy from dstBuffer to srcBuffer and check the
+  content of the whole dstBuffer
+* Validate the order of the copies in one command buffer:
+  first copy from srcBuffer to a region of dstBuffer, then copy from another part of srcBuffer to
+  another region of dstBuffer which have overlaps with the region of dstBuffer in the first copy
+  and check the content of the whole dstBuffer to see the copies are done in correct order.
 `;
 
+import { poptions, params } from '../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
 import { GPUTest } from '../../../gpu_test.js';
 
 export const g = makeTestGroup(GPUTest);
 
-g.test('b2b').fn(async t => {
-  const data = new Uint32Array([0x01020304]);
+g.test('b2b')
+  .params(
+    params()
+      .combine(poptions('srcOffset', [0, 4, 8, 16]))
+      .combine(poptions('dstOffset', [0, 4, 8, 16]))
+      .combine(poptions('copySize', [0, 4, 8, 16]))
+  )
+  .fn(async t => {
+    const { srcOffset, dstOffset, copySize } = t.params;
+
+    const kSrcBufferSizes = [srcOffset + copySize, srcOffset + copySize + 8];
```

Comment:
Maybe these should be in the parametrization so it's clear we iterate over them as well.

Reply:
I just don't want to test invalid parameters here. For example, srcBufferSize == 8 is not legal for the cases that "srcOffset + copySize > 8". Computing buffer sizes in this way can ensure the buffer size is legal in all the tests. What do you think?

Reply:
Maybe there's a way to put the computations in the params list? But if that's impractical let's ignore this comment.

Reply:
You can do something like this in the .combine()-chain:

```
      .expand(p => poptions('srcBufferSizes', [p.srcOffset + p.copySize, p.srcOffset + p.copySize + 8]))
      .expand(p => poptions('dstBufferSizes', [p.dstOffset + p.copySize, p.dstOffset + p.copySize + 8]))
```

Reply:
Done. Thanks @kainino0x  for your suggestions!